plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -203,30 +266,45 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                     if (severityFound.equals("warning")) {
                         markerSeverity = IMarker.SEVERITY_WARNING;
                     }
+                    if (severityFound.equals("note")) {
+                        markerSeverity = IMarker.SEVERITY_INFO;
+                    }
                     if (markerSeverity != -1) {
-                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2)));
+                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2))) - 1;
                         column = Integer.parseInt(outputLine.substring(m.start(3), m.end(3))) - 1;
                         messageId = "mypy";
                         message = outputLine.substring(m.start(5), m.end(5)).trim();
+
+                        IRegion region = null;
+                        try {
+                            region = document.getLineInformation(line);
+                        } catch (Exception e) {
+                        }
+                        if (region != null && document != null) {
+                            LineCol lineCol = new LineCol(line, column);
+                            if (lineColToMessage.containsKey(lineCol)) {
+                                lineColToMessage.get(lineCol).addMessageLine(message);
+                            } else {
+                                MessageInfo messageInfo = new MessageInfo(message, markerSeverity, messageId, line,
+                                        column, document.get(region.getOffset(), region.getLength()));
+                                lineColToMessage.put(lineCol, messageInfo);
+                            }
+                        }
                     } else {
                         continue;
                     }
                 } else {
                     continue;
                 }
-                IRegion region = null;
-                try {
-                    region = document.getLineInformation(line - 1);
-                } catch (Exception e) {
-                }
-                if (region != null && document != null) {
-                    String docLineContents = document.get(region.getOffset(), region.getLength());
-                    addToMarkers(message, markerSeverity, messageId, line - 1, column, docLineContents);
-                }
             } catch (Exception e) {
                 Log.log(e);
             }
         }
+        for (MessageInfo messageInfo : lineColToMessage.values()) {
+            addToMarkers(messageInfo.message, messageInfo.markerSeverity, messageInfo.messageId, messageInfo.line,
```

Comment:
Note: after changing the message to a `FastStringBuffer`, this will need a `toString()`.